pkg/proxy/apis/kubeproxyconfig/fuzzer/fuzzer.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuzzer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/gofuzz"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/kubernetes/pkg/kubelet/qos"
+	"k8s.io/kubernetes/pkg/master/ports"
+	"k8s.io/kubernetes/pkg/proxy/apis/kubeproxyconfig"
+	utilpointer "k8s.io/kubernetes/pkg/util/pointer"
+)
+
+// Funcs returns the fuzzer functions for the kube-proxy apis.
+func Funcs(codecs runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		func(obj *kubeproxyconfig.KubeProxyConfiguration, c fuzz.Continue) {
+			c.FuzzNoCustom(obj)
+			obj.BindAddress = "0.0.0.0"
+			obj.HealthzBindAddress = fmt.Sprintf("0.0.0.0:%v", ports.ProxyHealthzPort)
+			obj.MetricsBindAddress = fmt.Sprintf("127.0.0.1:%v", ports.ProxyStatusPort)
+			obj.OOMScoreAdj = utilpointer.Int32Ptr(int32(qos.KubeProxyOOMScoreAdj))
+			obj.ResourceContainer = "/kube-proxy"
+			obj.IPTables.SyncPeriod = metav1.Duration{Duration: 30 * time.Second}
+			obj.IPVS.SyncPeriod = metav1.Duration{Duration: 30 * time.Second}
+			obj.UDPIdleTimeout = metav1.Duration{Duration: 250 * time.Millisecond}
+			obj.Conntrack.MaxPerCore = utilpointer.Int32Ptr(32 * 1024)
+			obj.Conntrack.Min = utilpointer.Int32Ptr(128 * 1024)
+			obj.IPTables.MasqueradeBit = utilpointer.Int32Ptr(14)
+			obj.Conntrack.TCPEstablishedTimeout = &metav1.Duration{Duration: 24 * time.Hour}
+			obj.Conntrack.TCPCloseWaitTimeout = &metav1.Duration{Duration: 1 * time.Hour}
+			obj.ConfigSyncPeriod.Duration = 15 * time.Minute
+			obj.ClientConnection.ContentType = "application/vnd.kubernetes.protobuf"
+			obj.ClientConnection.QPS = 5.0
+			obj.ClientConnection.Burst = 10
```

Comment:
very static fuzzer :)

Reply:
https://xkcd.com/221/

Reply:
Hmm, these values are from defaulters. I should assign them "random" values here, right? :)

Reply:
Fuzzers use random values usually to catch encoding errors.

Reply:
Thanks for clarifying!

Reply:
Agreed.
A round trip test with at least defaults is better than nothing.
A round trip test that can assign random valid values within the constraints of each field is even better.
For kubeletconfig we have the former, and need to extend to the latter.

Reply:
`c.FuzzNoCustom(obj)` writes random values. Do we have to override them with constant values?

Reply:
>c.FuzzNoCustom(obj) writes random values. Do we have to override them with constant values?

@mtaufen any thoughts?